sdk/common/src/main/scala/org/typelevel/otel4s/sdk/Attributes.scala

```diff
@@ -75,32 +66,153 @@ final class Attributes private (
 
 object Attributes {
 
-  val Empty = new Attributes(Map.empty)
+  /** A '''mutable''' builder of [[Attributes]].
+    */
+  sealed trait Builder {
+
+    /** Adds the given `attribute` to the builder.
+      *
+      * @note
+      *   if the key of the given `attribute` is already present in the builder,
+      *   the value will be overwritten with the corresponding given attribute.
+      *
+      * @param attribute
+      *   the attribute to add
+      */
+    def addOne[A](attribute: Attribute[A]): Builder
+
+    /** Adds the attribute with the given `key` and `value` to the builder.
+      *
+      * @note
+      *   if the given `key` is already present in the builder, the value will
+      *   be overwritten with the given `value`.
+      *
+      * @param key
+      *   the key of the attribute. Denotes the types of the `value`
+      *
+      * @param value
+      *   the value of the attribute
+      */
+    def addOne[A](key: AttributeKey[A], value: A): Builder
 
-  def apply(attributes: Attribute[_]*): Attributes = {
+    /** Adds the given `attributes` to the builder.
+      *
+      * @note
+      *   if the keys of the given `attributes` are already present in the
+      *   builder, the values will be overwritten with the corresponding given
+      *   attributes.
+      *
+      * @param attributes
+      *   the attributes to add
+      */
+    def addAll(attributes: Attributes): Builder
+
+    /** Creates [[Attributes]] with the attributes of this builder.
+      */
+    def result(): Attributes
+  }
+
+  val Empty: Attributes = new MapAttributes(Map.empty)
+
+  /** Creates [[Attributes]] with the given `attributes`.
+    *
+    * @note
+    *   if there are duplicated keys in the given `attributes`, only the last
+    *   occurrence will be retained.
+    *
+    * @param attributes
+    *   the attributes to use
+    */
+  def apply(attributes: Attribute[_]*): Attributes =
+    fromIterable(attributes)
+
+  /** Creates [[Attributes]] from the given collection of `attributes`.
+    *
+    * @note
+    *   if there are duplicated keys in the given `attributes`, only the last
+    *   occurrence will be retained.
+    *
+    * @param attributes
+    *   the attributes to use
+    */
+  def fromIterable(attributes: Iterable[Attribute[_]]): Attributes = {
     val builder = Map.newBuilder[AttributeKey[_], Attribute[_]]
     attributes.foreach { a =>
       builder += (a.key -> a)
     }
-    new Attributes(builder.result())
+    new MapAttributes(builder.result())
   }
 
+  /** Creates [[Attributes]] from the given map of `attributes`.
+    *
+    * @param attributes
+    *   the attributes to use
+    */
+  def fromMap(attributes: Map[AttributeKey[_], Attribute[_]]): Attributes =
```

Comment:
I do not like this method being public, as it's fundamentally unsafe

Reply:
Valid point. It's not needed anywhere yet, so I don't see any reason to keep it.